packages/language-server/src/codeActionProvider.ts

```diff
@@ -0,0 +1,189 @@
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+  CodeActionParams,
+  CodeAction,
+  Diagnostic,
+  DiagnosticSeverity,
+  CodeActionKind,
+  Range,
+} from 'vscode-languageserver'
+import { isNullOrUndefined } from 'util'
+import { getAllRelationNames } from './completion/completions'
+import { convertDocumentTextToTrimmedLineArray } from './MessageHandler'
+
+function insertModelOrEnumAtRange(document: TextDocument): Range {
```

Comment:
I suggest `getInsertRange` or something more generic as a name for this function as it has nothing to do with either model or enum.